#!/usr/bin/env python
PACKAGE = "external_parameter_adjustment"
from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator() # create a object of parameter generator
#gen.add(parameter_name, parameter_type, bit_mask, description, default_value, minimum, maximum)
gen.add("translation_x", double_t, 0, "translation of x axis", 0, -20, 20)
gen.add("translation_y", double_t, 0, "translation of y axis", 0, -20, 20)
gen.add("translation_z", double_t, 0, "translation of z axis", 0, -20, 20)
gen.add("rotation_x", double_t, 0, "rotation angle of x axis", 0, -180, 180)
gen.add("rotation_y", double_t, 0, "rotation angle of y axis", 0, -180, 180)
gen.add("rotation_z", double_t, 0, "rotation angle of z axis", 0, -180, 180)

exit(gen.generate(PACKAGE, "parameter_adjustment_node", "parameter"))
#first parameter is namespace's name
#second parameter is node's name
#third parameter is the current file's name